### ISSUE TITLE
Support for `ScaleInvariant` Loss

### ISSUE THREAD

**Issue by awsaf49** [keras-team/tf-keras#229]
**System information**.

TensorFlow version (you are using): 2.11.0
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.
The Scale-Invariant (SI) loss is very popular in the field of Depth Estimation introduced in [this paper](https://proceedings.neurips.cc/paper/2014/file/7bccfde7714a1ebadf06c5f4cea752c1-Paper.pdf). Almost all papers in Depth Estimation use this loss for the training models. The mathematical equation for this loss is:

$$
\mathcal{L}_{\text {pixel }}=\alpha \sqrt{\frac{1}{T} \sum_{i} g_i^2-\frac{\lambda}{T^2}\left(\sum_i g_i\right)^2}
$$

Where $d_i$ is the ground truth depth, $\hat{d}_i$  is the estimated depth, $T$ represents the number of pixels with valid ground truth values and $g_i$ is computed as $g_i=\log_e(\hat{d}_i)-\log_e(d_i)$. The parameters $\lambda$ and $\alpha$ are hyperparameter.

**Will this change the current api? How?** N/A

**Who will benefit from this feature?**
Scale Invariant loss is a regression loss that can be applied to any pixel-wise regression task. Specifically, this loss is widely used in Depth Estimation thus anyone who is working in this field will be benefited from it.

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**

- Do you want to contribute a PR? (yes/no): Yes
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

Here is a candidate solution

```py
class ScaleInvariant(tf.keras.losses.Loss):
    def __init__(self,
                 alpha=10.0,
                 lamda=0.85,
                ):
        super().__init__()
        self.alpha = alpha
        self.lamda = lamda

    def call(self, y_true, y_pred):
        # create vaid mask
        mask = self.get_mask(y_true)
        # compute loss
        loss = self.compute_loss(y_true, y_pred, mask)
        return loss
```
